### PR TITLE
Use lib crates from crates.io

### PR DESCRIPTION
**Pull request by davidlattimore**
See lib/README.md for justification

### CODE REVIEW

**Comment by davidlattimore** [comment]
This should help reduce breakages to the autopublish workflow.  #15038

File: Cargo.toml [hunk 1]
```diff
@@ -76,10 +76,10 @@ tt = { path = "./crates/tt", version = "0.0.0" }
...
 vfs = { path = "./crates/vfs", version = "0.0.0" }
 
-# In-tree crates that are published separately and follow semver.
-line-index = { version = "0.1.0-pre.1", path = "lib/line-index" }
-la-arena = { version = "0.3.1", path = "lib/la-arena" }
-lsp-server = { version = "0.7.1", path = "lib/lsp-server" }
+# In-tree crates that are published separately and follow semver. See lib/README.md
+line-index = { version = "0.1.0-pre.1" }
+la-arena = { version = "0.3.1" }
+lsp-server = { version = "0.7.1" }
```
Comment (Veykril) [inline Cargo.toml:82]: Up above is a `[patch.'crates-io']` section with a lot of commented out patches, could you  add commented out path versions for these there? That would make it easier to use the local ones on demand for testing.

Comment (davidlattimore) [reply Cargo.toml:86]: Good idea. Done. I also amended the last bit that I'd added to libs/README.md accordingly.

**Comment by Veykril** [comment]
Thanks
@bors r+

**Pull request merged**